sympy/stats/drv.py

```diff
@@ -30,24 +31,29 @@ def check(*args):
     def sample(self):
         """ A random realization from the distribution """
         icdf = self._inverse_cdf_expression()
-        return floor(icdf(random.uniform(0, 1)))
+        while(True):
+            sample_ = floor(list(icdf(random.uniform(0, 1)))[0])
+            if sample_ >= self.set.inf:
+                return sample_
 
     @cacheit
     def _inverse_cdf_expression(self):
         """ Inverse of the CDF
 
         Used by sample
         """
-        x, z = symbols('x, z', real=True, positive=True, cls=Dummy)
+        x = symbols('x', positive = True,
+         integer = True, cls=Dummy)
+        y, z = symbols('y z', positive=True, cls=Dummy)
+        cdf_temp = self.cdf(x).subs(x, y)
```

Comment:
Still, why `self.cdf(x).subs(x, y)` here instead of `self.cdf(y)`? I see no reason to insert one variable just to immediately replace it with another.

Reply:
@normalhuman Sorry, you are right. I was substituting the variables to remove the integer constraint from the `cdf` variable, in order to help `solveset`. But `solveset` works fine either way. I have made the necessary changes.